tests/beaker_tests/cisco_interface_ospf/test_interface_ospf.rb

```diff
@@ -0,0 +1,126 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+###############################################################################
+#
+# See README-develop-beaker-scripts.md (Section: Test Script Variable Reference)
+# for information regarding:
+#  - test script general prequisites
+#  - command return codes
+#  - A description of the 'tests' hash and its usage
+#
+###############################################################################
+
+require File.expand_path('../../lib/utilitylib.rb', __FILE__)
+
+# Test hash top-level keys
+tests = {
+  master:           master,
+  agent:            agent,
+  intf_type:        'ethernet',
+  operating_system: 'nexus',
+  resource_name:    'cisco_interface_ospf',
+}
+
+# Skip -ALL- tests if a top-level platform/os key exludes this platform
+skip_unless_supported(tests)
+
+# Find a usable interface for this test
+@intf = find_interface(tests)
+tp = @intf + ' Sample'
+
+# Test hash test cases
+tests[:default] = {
+  desc:           '1.1 Defaults',
+  title_pattern:  tp,
+  preclean_intf:  true,
+  manifest_props: {
+    area:                           200,
+    cost:                           'default',
+    dead_interval:                  'default',
+    hello_interval:                 'default',
+    message_digest:                 'default',
+    message_digest_key_id:          'default',
+    message_digest_algorithm_type:  'default',
+    message_digest_encryption_type: 'default',
+    message_digest_password:        'default',
+    passive_interface:              'default',
+  },
+  code:           [0, 2],
+  resource:       {
+    area:                           '0.0.0.200',
+    cost:                           0,
+    dead_interval:                  40,
+    hello_interval:                 10,
+    message_digest:                 'false',
+    message_digest_key_id:          0,
+    message_digest_algorithm_type:  'md5',
+    message_digest_encryption_type: 'cleartext',
+    passive_interface:              'false',
+  },
+}
+
+# Non-default Tests. NOTE: [:resource] = [:manifest_props] for all non-default
+
+tests[:non_default] = {
+  desc:           '2.1 Non Defaults',
+  title_pattern:  tp,
+  preclean_intf:  true,
+  manifest_props: {
+    area:                           200,
+    cost:                           '200',
+    dead_interval:                  '200',
+    hello_interval:                 '200',
+    message_digest:                 'true',
+    message_digest_key_id:          27,
+    message_digest_algorithm_type:  'md5',
+    message_digest_encryption_type: 'cisco_type_7',
+    message_digest_password:        '046E1803362E595C260E0B240619050A2D',
+    passive_interface:              'true',
+  },
+  resource:       {
+    area: '0.0.0.200'
+  },
+}
+
+# Overridden to properly handle dependencies for this test file.
+def test_harness_dependencies(_tests, _id)
```

Comment:
It looks like this really only needs to process at the beginning of the test. If so just add an early return:

`  return unless id == :default`